src/App.tsx

```diff
@@ -8,7 +8,52 @@ import { TodoFilter } from './components/TodoFilter';
 import { TodoModal } from './components/TodoModal';
 import { Loader } from './components/Loader';
 
+import { getTodos } from './api';
+import { Todo } from './types/Todo';
+import { filterTodos } from './helpers';
+import { Filter } from './types/Filter';
+
 export const App: React.FC = () => {
+  const [isLoading, setIsLoading] = useState(true);
+  const [todosFromServer, setTodosFromServer] = useState<Todo[]>([]);
+  const [selectedTodo, setSelectedTodo] = useState<Todo | null>(null);
+  const [filter, setFilter] = useState<Filter>(Filter.All);
+  const [query, setQuery] = useState('');
+
+  useEffect(() => {
+    getTodos()
+      .then(result => {
+        setTodosFromServer(result);
+        setIsLoading(false);
+      });
+  }, []);
+
+  const visibleTodos = query
+    ? filterTodos(todosFromServer, Filter.All, query)
+    : filterTodos(todosFromServer, filter, '');
```

Comment:
query is `''` in its default state, so it seems to me this variable with ternary is redundant